lib/client.dart

```diff
@@ -1,59 +1,128 @@
 import 'dart:async';
 import 'dart:convert';
+
+import 'package:gql/execution.dart' as exec;
+import 'package:gql/language.dart';
+import 'package:gql/link.dart' as gql_link;
 import 'package:http/http.dart' as http;
 import 'package:json_annotation/json_annotation.dart';
+
 import './schema/graphql_error.dart';
 import './schema/graphql_query.dart';
 
-class _DefaultHttpJsonClient extends http.BaseClient {
+/// Simple implementation of an HttpLink
+class SimpleHttpJsonLink implements gql_link.Link {
+  /// Endpoint of a GraphQL service
+  final String graphQLEndpoint;
+
   http.Client _httpClient = http.Client();
 
-  _DefaultHttpJsonClient();
+  /// Construct the Link
+  SimpleHttpJsonLink(this.graphQLEndpoint);
```

Comment:
As we'll already have `SimpleHttpJsonLink` to use as a default for `ArtemisClient`'s link, I think it should optionally receive a `http.Client` on its constructor (so it's easier to inject credentials and tokens, for example).

Reply:
That makes sense for compatibility reasons.

Reply:
Agree that receiving the client is an important feature (or we would need to write a custom Link just to be able to wrap our client)